tests/integration/test_sqs.py

```diff
@@ -174,4 +176,44 @@ def receive_dlq():
             self.assertIn('RequestID', msg_attrs)
             self.assertIn('ErrorCode', msg_attrs)
             self.assertIn('ErrorMessage', msg_attrs)
+
         retry(receive_dlq, retries=8, sleep=2)
+
+    def test_set_queue_attribute_at_creation(self):
+        queue_name = 'queue-%s' % short_uid()
+
+        attributes = {
+            'MessageRetentionPeriod': '604800',  # This one is unsupported by ElasticMq and should be saved in memory
+            'ReceiveMessageWaitTimeSeconds': '10',
+            'VisibilityTimeout': '30'
+        }
+
+        queue_url = self.client.create_queue(QueueName=queue_name, Attributes=attributes)['QueueUrl']
+        creation_attributes = self.client.get_queue_attributes(QueueUrl=queue_url, AttributeNames=['All'])
+
+        # assertion
+        self.assertIn('MessageRetentionPeriod', creation_attributes['Attributes'].keys())
```

Comment:
nitpick: We could also assert that the correct value is returned here (`604800`). Same for the test `test_get_specific_queue_attribute_response` below.